### DISCUSSION TITLE
Add a generic filter property to define custom raw sql WHERE conditions

### DISCUSSION THREAD

**Discussion by RaduCeban** [mikro-orm/mikro-orm#5068] [General]
I'm troubled by an implementation in my project that heavily relies on EntityManager.find helper function, but I need to pass on some raw sql filters that can't be built by existing operations  (like $gt, $gte, $eq, etc), so maybe it'd make sense to add some prop to allow that. For example: 

Given this raw sql condition: 
**(review_completed >= (CURRENT_TIMESTAMP - INTERVAL '24 hours') OR review_completed IS NULL) AND deleted IS NOT TRUE**
I'd want it to result in:
**where (review_completed >= (CURRENT_TIMESTAMP - INTERVAL '24 hours') OR review_completed IS NULL) AND deleted IS NOT TRUE**

But not:

**where = (review_completed >= (CURRENT_TIMESTAMP - INTERVAL '24 hours') OR review_completed IS NULL) AND deleted IS NOT TRUE**

As a workaround I created this custom filter, but it's not quite the same:
`@Filter({
  name: 'rw',
  args: false,
  default: true,
  cond: (_, type, em: EntityManager) => {
    if (type != 'read') return {}; // do not apply filter

    return {
      review_completed: {
        $gte: em.raw("(CURRENT_TIMESTAMP - INTERVAL '24 hours') OR review_completed IS NULL AND deleted IS NOT TRUE")
      },
    };
  },
})`

I'd assume the resulted where clause would be: 
**review_completed >= (CURRENT_TIMESTAMP - INTERVAL '24 hours') OR review_completed IS NULL AND deleted IS NOT TRUE**

But, this operation won't take precedence as I'm intending to do: 

**(review_completed >= (CURRENT_TIMESTAMP - INTERVAL '24 hours') OR review_completed IS NULL)**

**Comment by B4nan** [comment]
You forgot to mention the code you use with `em.find`, I don't see how it could behave differently from filters (as filters are just about altering the where condition automatically).

>  review_completed: {
>    $gte: em.raw("(CURRENT_TIMESTAMP - INTERVAL '24 hours') OR review_completed IS NULL AND deleted IS NOT TRUE")
>  },

This looks weird as you are mixing the value for the `$gte` operator with another OR condition, you should split that or use a single raw query, not this mixture, which is what probably introduces your problem. Try something like this:

```ts
await em.find(T, {
  $or: [
    {
      review_completed: {
        $gte: em.raw("(CURRENT_TIMESTAMP - INTERVAL '24 hours')"),
      },
    },
    {
      review_completed: null,
    },
  ],
  deleted: { $not: true }, // this is also weird, why not just `deleted: false`? or it can be null as well? thats a weird design too :]
});
```

(not sure what exact query you want, since you mix AND and OR in one expression, you might want to restructure this)
